### PR TITLE
Andreas/hail mary

### PR DESCRIPTION
**Pull request by akasprzok**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
-    - uses: ./
-      with:
-        version: 'latest'
+      - uses: actions/checkout@v2
+      - uses: ./
+        with:
+          file: "mass-v0.4.8-linux-amd64.tar.gz"
```
Comment (coryodaniel) [inline .github/workflows/test.yml:24]: I think one of the fields that we can expose on the input is the version and we can default it to some number

File: action.yml [hunk 2]
```diff
@@ -1,10 +1,18 @@
-name: 'Massdriver Setup Action'
-description: 'Sets up the Massdriver CLI'
+name: 'Your name here'
+description: 'Provide a description here'
+author: 'Your name or organization here'
 inputs:
-  version: 
-    description: 'The version of the Massdriver CLI to download'
+  version:
```
Comment (coryodaniel) [inline action.yml:5]: Nm above

File: src/main.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      headers: {
+        accept: 'application/octet-stream',
+      },
+      owner,
+      repo,
+    }
+  );
+  let headers: HeadersInit = {
+    accept,
+    authorization: `token ${token}`,
```
Comment (coryodaniel) [inline src/main.ts:81]: I’m not positive you need a token to download the cli since it’s public

**Review by coryodaniel — APPROVED** [review]

File: src/main.ts [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  });
+
+const printOutput = (release: GetReleaseResult): void => {
+  core.info(`version: ${release.data.tag_name}`);
+  core.info(`name: ${release.data.name}`);
+};
+
+const install = async (target: string) => {
+  core.info(`target: ${target}`);
+  const pathToCLI = await tc.extractZip(target);
```
Comment (akasprzok) [inline src/main.ts:117]: This currently fails

File: .github/workflows/test.yml (hunk 1 above)
Comment (akasprzok) [reply .github/workflows/test.yml:24]: Yeah, that'd be ideal. Still trying to get there.

**Pull request merged**